src/breadboard-element.ts

```diff
@@ -0,0 +1,328 @@
+import { css, customElement, html, LitElement, property, svg } from 'lit-element';
+
+@customElement('wokwi-breadboard')
+export class BreadboardElement extends LitElement {
+  @property() size: 'full' | 'half' | 'mini' = 'full';
+
+  static get styles() {
+    return css`
+      .container {
+        border-radius: 3px;
+        background: #eeefed;
+      }
+      .bottom {
+        transform: translate(0, 166px);
+      }
+      .spot {
+        font-size: 7px;
+        opacity: 0.6;
+      }
+    `;
+  }
+  // Next value is the sum of the prev value + the amount we want to add
+  private increaseBy(numToAdd: number, initValue: number, membersCount: number) {
+    const xPositions = [initValue];
+    for (let i = 1; i <= membersCount; i++) {
+      xPositions.push(xPositions[i - 1] + numToAdd);
+    }
+    return xPositions;
+  }
+
+  private getFullSized(increaseFn: (num: number, init: number, count: number) => number[]) {
+    const gapBetweenBlocks = increaseFn(55, 35, 10); // Will create a space between each pattern
+    const textPositions = increaseFn(45, 53, 13);
+    return html`
+      <svg width="163mm" height="53mm" xmlns="http://www.w3.org/2000/svg" class="container">
+        <defs>
+          <pattern id="points" width="2mm" height="2mm" patternUnits="userSpaceOnUse" x="3" y="5">
+            <rect width="1mm" height="1mm" fill-opacity="0.5" />
+          </pattern>
+          <pattern
+            id="points-wide"
+            width="2.5mm"
+            height="2.5mm"
+            patternUnits="userSpaceOnUse"
+            x="5.3"
+            y="5.5"
+          >
+            <rect width="1mm" height="1mm" fill-opacity="0.5" />
+          </pattern>
+        </defs>
+
+        <g class="top">
+          <g class="plus" transform="scale(0.5)">
+            <line x1="4" x2="17" y1="10.5" y2="10.5" stroke="#F97466" stroke-width="1" />
+            <line y1="4" y2="17" x1="10.5" x2="10.5" stroke="#F97466" stroke-width="1" />
+          </g>
+          <g class="minus" transform="translate(0 23) scale(0.5)">
+            <line y1="4" y2="17" x1="10.5" x2="10.5" stroke="#8CE2F3" stroke-width="1" />
+          </g>
+          <rect x="6mm" y="5" width="149mm" height="1" fill-opacity="0.5" fill="#F97466" />
+          <rect
+            transform="translate(${gapBetweenBlocks[0]})"
+            y="12.55"
+            width="40"
+            height="3.6mm"
+            fill="url(#points)"
+          />
```

Comment:
These repeated `rect` elements can be replaced with iterating over the array using `map`, e.g.:

```typescript
${gapBetweenBlocks.map(gap => 
  svg`<rect
            transform="translate(${gap})"
            y="12.55"
            width="40"
            height="3.6mm"
            fill="url(#points)"
  />`)
}